opensearch-operator/pkg/reconcilers/cluster.go

```diff
@@ -548,3 +553,31 @@ func (r *ClusterReconciler) deleteSTSWithOrphan(existing *appsv1.StatefulSet) er
 	}
 	return nil
 }
+
+func (r *ClusterReconciler) UpdateClusterHealth() error {
+	health := util.GetClusterHealth(r.ctx, r.Client, r.instance)
+
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		if err := r.Get(r.ctx, client.ObjectKeyFromObject(r.instance), r.instance); err != nil {
+			return err
+		}
+		r.instance.Status.Health = health
+		return r.Status().Update(r.ctx, r.instance)
+	})
+
+	return err
+}
+
+func (r *ClusterReconciler) UpdateAvailableNodes() error {
+	availableNodes := util.GetAvailableOpenSearchNodes(r.ctx, r.Client, r.instance)
+
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		if err := r.Get(r.ctx, client.ObjectKeyFromObject(r.instance), r.instance); err != nil {
+			return err
+		}
+		r.instance.Status.AvailableNodes = availableNodes
+		return r.Status().Update(r.ctx, r.instance)
+	})
+
+	return err
+}
```

Comment:
I'd prefer to have both status fields updated in one go, to reduce the number of API interactions.